kart/spatial_filters.py

```diff
@@ -105,26 +125,41 @@ def matches(self, feature_geometry):
 
     def transform_for_dataset(self, dataset):
         """Transform this spatial filter so that it matches the CRS of the given dataset."""
-        if self.match_all or self.filter_ogr is None or self.crs is None:
-            return self
+        if self.match_all or dataset is None:
```

Comment:
needs a comment why `dataset=None` is sensible

Reply:
Probably is not sensible - I thought needed it for doing diff(old_dataset, new_dataset, spatial_filter), which kind of turns into diff(old_dataset + old_spatial_filter, new_dataset + new_spatial_filter) - but old_dataset or new_dataset could be None, in which case, there's no particular spatial-filter that applies to them except MATCH_ALL (or MATCH_NONE would work too).
But I didn't end up structuring the code that way so never mind!